frontend/assets/javascripts/src/utils/user.js

```diff
@@ -12,39 +12,18 @@ define([
     };
 
     var getUserFromCookie = function(){
-        var userFromCookieCache = null;
-
-        function readCookie(name){
-            var nameEQ = name + '=';
-            var ca = document.cookie.split(';');
-            for (var i = 0; i < ca.length; i++) {
-                var c = ca[i];
-                while (c.charAt(0) === ' ') { c = c.substring(1, c.length); }
-                if (c.indexOf(nameEQ) === 0) { return c.substring(nameEQ.length, c.length); }
-            }
-            return null;
-        }
-
-        function decodeBase64(str){
-            /*global escape: true */
-            /* See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape */
-            return decodeURIComponent(escape(new AtoB()(str.replace(/-/g, '+').replace(/_/g, '/').replace(/,/g, '='))));
+        var userFromCookieCache;
+        var cookieData = cookie.getCookie('GU_U'),
+            userData = cookie.decodeCookie(cookieData);
+        if (userData) {
+            userFromCookieCache = {
+                id: userData[0],
+                displayname: userData[2],
+                accountCreatedDate: userData[6],
+                emailVerified: userData[7],
```

Comment:
are we always guaranteed to have these in userData? It may be a little overkill but maybe 
``userData[0] && userData[0]`` could be used as assignments to the relevant objects properties?